From 9b03ca4bb5c1868826a508169c38180df9bbb4a5 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:46 -0400
Subject: [tools] perf: Call perf_hpp__init() before setting up GUI browsers

Message-id: <1410181752-25631-222-git-send-email-jolsa@redhat.com>
Patchwork-id: 91897
O-Subject: [PATCH RHEL7.1 BZ1134356 221/307] perf tools: Call perf_hpp__init() before setting up GUI browsers
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 22af969e8cfc6ea46d3e1a774a16d7e19b8cf4db
Author: Namhyung Kim <namhyung@kernel.org>
Date: Wed Apr 16 11:04:51 2014 +0900

description
===========
So that it can be set properly prior to set up output fields.  That
makes easy to handle/warn errors during the setup since it doesn't
need to be bothered with the GUI.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index f4d640c..c4dab7a 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -823,16 +823,16 @@ repeat:
    goto error;
  }
 
+ perf_hpp__init();
+
  /* Force tty output for header output. */
  if (report.header || report.header_only)
   use_browser = 0;
 
  if (strcmp(input_name, "-") != 0)
   setup_browser(true);
- else {
+ else
   use_browser = 0;
-  perf_hpp__init();
- }
 
  if (report.header || report.header_only) {
   perf_session__fprintf_info(session, stdout,
diff --git a/tools/perf/builtin-top.c b/tools/perf/builtin-top.c
index 34764b6..280945b 100644
--- a/tools/perf/builtin-top.c
+++ b/tools/perf/builtin-top.c
@@ -1147,6 +1147,8 @@ int cmd_top(int argc, const char **argv, const char *prefix __maybe_unused)
  /* display thread wants entries to be collapsed in a different tree */
  sort__need_collapse = 1;
 
+ perf_hpp__init();
+
  if (top.use_stdio)
   use_browser = 0;
  else if (top.use_tui)
diff --git a/tools/perf/ui/browsers/hists.c b/tools/perf/ui/browsers/hists.c
index 37c5188..92d128f 100644
--- a/tools/perf/ui/browsers/hists.c
+++ b/tools/perf/ui/browsers/hists.c
@@ -661,8 +661,6 @@ __HPP_COLOR_PERCENT_FN(overhead_guest_us, period_guest_us)
 
 void hist_browser__init_hpp(void)
 {
- perf_hpp__init();
-
  perf_hpp__format[PERF_HPP__OVERHEAD].color =
     hist_browser__hpp_color_overhead;
  perf_hpp__format[PERF_HPP__OVERHEAD_SYS].color =
diff --git a/tools/perf/ui/gtk/hists.c b/tools/perf/ui/gtk/hists.c
index 2237245..fd52669 100644
--- a/tools/perf/ui/gtk/hists.c
+++ b/tools/perf/ui/gtk/hists.c
@@ -58,8 +58,6 @@ __HPP_COLOR_PERCENT_FN(overhead_guest_us, period_guest_us)
 
 void perf_gtk__init_hpp(void)
 {
- perf_hpp__init();
-
  perf_hpp__format[PERF_HPP__OVERHEAD].color =
     perf_gtk__hpp_color_overhead;
  perf_hpp__format[PERF_HPP__OVERHEAD_SYS].color =
diff --git a/tools/perf/ui/setup.c b/tools/perf/ui/setup.c
index 5df5140..ba51fa8 100644
--- a/tools/perf/ui/setup.c
+++ b/tools/perf/ui/setup.c
@@ -86,8 +86,6 @@ void setup_browser(bool fallback_to_pager)
   use_browser = 0;
   if (fallback_to_pager)
    setup_pager();
-
-  perf_hpp__init();
   break;
  }
 }
-- 
1.7.1